src/Tree.jsx

```diff
@@ -322,6 +324,10 @@ class Tree extends React.Component {
     const eventKey = treeNode.props.eventKey;
     const selected = !treeNode.props.selected;
 
+    if (props.checkable && props.checkOnSelect) {
+      this.onCheck(treeNode);
```

Comment:
Actually, `this.onCheck` is toggle check https://github.com/react-component/tree/blob/master/src/Tree.jsx#L268

So, in your code, if a user check the checkbox with clicking on checkbox, and then click the label, it will uncheck the checkbox.

Reply:
@benjycui That is the behavior that we should have. So if you have a checkbox, clicking label or clicking the box, will do the same thing - check or uncheck the checkbox depending on its current state. It is very clumsy not to be able to click on the label for check/uncheck. I still think better solution would be not to even have an option for user and force checkOnSelect if it is checkable. 

@afc163 mentioned ant-design/ant-design#7303. This PR is exactly for that issue.